src/merkle/tiered_smt/mod.rs

```diff
@@ -38,7 +38,7 @@ pub struct TieredSmt {
     nodes: BTreeMap<NodeIndex, RpoDigest>,
     upper_leaves: BTreeMap<NodeIndex, RpoDigest>, // node_index |-> key map
     bottom_leaves: BTreeMap<u64, BottomLeaf>,     // leaves of depth 64
-    values: BTreeMap<RpoDigest, Word>,
+    values: BTreeMap<RpoDigest, RpoDigest>,
```

Comment:
I'm thinking we should keep `values` (and related interfaces) as `BTreeMap<RpoDigest, Word>` (basically, the way it was). Would it complicate things?